### ISSUE TITLE
Improve messaging when baseline for a given file is exceeded

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#1062]
Causing some confusion locally.

We could store the snippet selection text for each error in the baseline - it would make the file larger, but it would make false-positives much less likely.

cc @ErikBooijCB

**Comment by ErikBooijCB** [comment]
Interesting idea indeed.

The file would look something like this?

```xml
<?xml version="1.0" encoding="UTF-8"?>
<files>
  <file src="src/folder/File1.php">
    <MoreSpecificImplementedParamType>
       ... code snippet
    </MoreSpecificImplementedParamType>
  </file>
  <file src="src/folder/File1.php">
    <PossiblyInvalidMethodCall>
       ... code snippet
    </PossiblyInvalidMethodCall>
  </file>
</files
```

And when checking, compare the type *and* snippet?

I can look into that, sometime soon.

**Comment by muglug** [comment]
Maybe more like 

```xml
<?xml version="1.0" encoding="UTF-8"?>
<files>
  <file src="src/folder/File1.php">
    <MoreSpecificImplementedParamType occurrences="2">
       <code><!-- function fooBar() --></code>
    </MoreSpecificImplementedParamType>
  </file>
  <file src="src/folder/File1.php">
    <PossiblyInvalidMethodCall>
       ... code snippet
    </PossiblyInvalidMethodCall>
  </file>
</files>
```

Any code with a line break would be omitted (most don't have line breaks)

**Comment by weirdan** [comment]
> Any code with a line break would be omitted (most don't have line breaks)

@muglug I wonder why would we need to omit those? I'm observing funny behaviour here, which I suppose is what this ticket was about, for multiline calls to a deprecated function like this:
```php
<?php
/** @deprecated */
function send_mail(string $_subject, string $_body): void {}

send_mail( // newly added, but not flagged
    'Another subj',
    'Another message'
);

send_mail( // grandfathered in in baseline, but is flagged 
    'Subj', 
    'Message'
);
```
